components/lib/dropdown/Dropdown.js

```diff
@@ -827,7 +827,7 @@ export const Dropdown = React.memo(
         const getOptionValue = (option) => {
             const optionValue = props.optionValue ? ObjectUtils.resolveFieldData(option, props.optionValue) : option ? option['value'] : ObjectUtils.resolveFieldData(option, 'value');
 
-            return ObjectUtils.isNotEmpty(optionValue) ? optionValue : option;
+            return props.optionValue !== undefined ? optionValue : option;
```

Comment:
```suggestion
            return optionValue !== undefined ? optionValue : option;
```

Reply:
props.optionValue should also be checked for undefined in line before:)